pymeasure/instruments/smaract/utils.py

```
@@ -0,0 +1,52 @@
+def set_type(response_raw: str, index: int, unit: str = 'Hz'):
+    """Parse a device response substring into a Quantity with the given unit.
+
+    :param response_raw: Raw response string from the device.
+    :param index: Starting index for the substring to parse.
+    :param unit: Unit string for the resulting Quantity.
+    :returns: Quantity parsed from the response substring.
+    """
+    return Q_(response_raw[index:], unit)
+
+
+def convert_quantity_to_magnitude(input: Union[str, int, Q_], unit: str) -> Union[int, float]:
```

Ruff (0.15.12)

[error] Function argument `input` is shadowing a Python builtin

(A002)